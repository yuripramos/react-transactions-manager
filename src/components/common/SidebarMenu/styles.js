import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { white, trueWhite } from "../../../styles/settings";
import Icon from "../Icon";

export const MenuList = styled.ul`
  background: ${white};
  height: 100vh;
`;

export const Divisor = styled.div`
  height: ${rem(25)};
  box-shadow: 0 ${rem(10)} ${rem(15)} ${rem(6)} rgba(159, 189, 204, 0.15);
  margin-bottom: ${rem(25)};
`;

export const ListItem = styled.li`
  cursor: pointer;
  position: relative;
  height: ${rem(65)};
  font-family: Lato;
  font-size: ${rem(20)};
  color: ${trueWhite};
  letter-spacing: ${rem(1)};
  padding-top: ${rem(20)};
  margin: 0 ${rem(25)} 0 ${rem(70)};

  ${({ active }) =>
    active &&
    css`
      color: ${white};
    `};

  :hover {
    opacity: 0.7;
  }

`;

export const IconWrapper = styled.div`
  position: absolute;
  left: -${rem(45)};
  top: ${rem(16)};
`;

export const Item = styled.div`
  cursor: pointer;
  flex-flow: row wrap;
  width: ${rem(80)};
  text-align: center;
  position: relative;
  border-top: solid ${rem(2)} transparent;

  ${media.md(css`
    width: ${rem(130)};
  `)};

  ${({ active }) =>
    !active &&
    css`
      :hover {
        color: ${white};
      }
    `};

  ${({ active }) =>
    active &&
    css`
      cursor: initial;
      color: ${white};
      font-weight: 600;
      border-top: solid ${rem(2)} ${white};
    `};
`;

export const MenuIcons = styled(Icon)`
  width: ${rem(18)};
  padding-top: ${rem(10)};
  padding-bottom: ${rem(9)};
  display: flex;
  align-items: center;
  justify-content: center;
  ${media.sm(css`
    width: auto;
  `)};
`;

export const Span = styled.span`
  margin-top: ${rem(9)};
  font-size: ${rem(12)};
`;

export const ListMenuItems = styled.div`
  margin: 0 auto;
  font-weight: 500;
  color: ${trueWhite};
  margin-top: ${rem(15)};
  display: flex;
  flex-flow: row wrap;
  height: ${rem(30)};
`;
