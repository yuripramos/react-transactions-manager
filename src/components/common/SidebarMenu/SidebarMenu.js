import React, { Component } from "react";
import { redirect } from "../../../utils/redirect";
import { ListMenuItems, Item, Span } from "./styles";


class SidebarMenu extends Component {

  constructor(props) {
    super(props);

    this.redirectTo = this.redirectTo.bind(this);
    this.isActive = this.isActive.bind(this);
  }

  redirectTo(path) {
    redirect(path);
  }

  isActive(path) {
    const { currentLocation } = this.props;
    return currentLocation.includes(path);
  }
  render() {
    return (
      <ListMenuItems>
        <Item
        >
          <Span>SHIPPER</Span>
        </Item>
        <Item
        >
          <Span>BECOME A CARRIER</Span>
        </Item>
        <Item
        >
          <Span>LOGIN</Span>
        </Item>
        <Item
        >
          <Span>SIGN UP</Span>
        </Item>
      </ListMenuItems>
    );
  }
}

export default SidebarMenu;
