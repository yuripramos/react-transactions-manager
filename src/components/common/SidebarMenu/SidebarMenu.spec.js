import React from "react";
import { shallow } from "enzyme";

import SidebarMenu  from "./SidebarMenu";



const props = {
  currentLocation: ""
};

describe("SidebarMenu component", () => {
  it("should match snapshot", () => {
    expect(shallow(<SidebarMenu {...props} />)).toMatchSnapshot();
  });
});

