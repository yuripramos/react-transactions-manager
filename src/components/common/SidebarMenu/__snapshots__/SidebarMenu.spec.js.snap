// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu component should match snapshot 1`] = `
<styled.div>
  <styled.div
    active={false}
    data-test="Navbar_politics"
    onClick={[Function]}
  >
    <styled.span>
      Politics
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    onClick={[Function]}
  >
    <styled.span>
      World
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_technology"
    onClick={[Function]}
  >
    <styled.span>
      Technology
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_health"
    onClick={[Function]}
  >
    <styled.span>
      Health
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_science"
    onClick={[Function]}
  >
    <styled.span>
      Science
    </styled.span>
  </styled.div>
</styled.div>
`;

exports[`SidebarMenu component should match snapshot for desktop 1`] = `
<styled.div>
  <styled.div
    active={false}
    data-test="Navbar_politics"
    onClick={[Function]}
  >
    <styled.span>
      Politics
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    onClick={[Function]}
  >
    <styled.span>
      World
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_technology"
    onClick={[Function]}
  >
    <styled.span>
      Technology
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_health"
    onClick={[Function]}
  >
    <styled.span>
      Health
    </styled.span>
  </styled.div>
  <styled.div
    active={false}
    data-test="Navbar_science"
    onClick={[Function]}
  >
    <styled.span>
      Science
    </styled.span>
  </styled.div>
</styled.div>
`;
